packages/modules/common/abstract_vehicle.py

```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class VehicleUpdateData:
+    plug_state: bool = False
+    charge_state: bool = False
+    imported: float = 0
+    battery_capacity: float = 82
+    soc_from_cp: float = 0
+
+
+@dataclass
+class GeneralVehicleConfig:
+    efficiency: float = 90
+    use_soc_from_cp: bool = False
+    request_interval_charging: int = 5
+    request_interval_not_charging: int = 720
+    request_only_plugged: bool = False
+    imported_start: Optional[float] = 0  # don't show in UI
+    manual_soc: Optional[int] = None  # don't show in UI
+    request_start_soc: bool = True  # don't show in UI
+    soc_start: float = 0  # don't show in UI
```

Comment:
Die Daten `imported_start`, `manual_soc`, `soc_start` und `request_start_soc` sind eigentlich keine Konfigurationsdaten, sondern Bewegungsdaten.
Alles in der Konfiguration sollte einmalig eingerichtet werden und danach statisch bleiben. Diese vier Variablen werden jedoch regelmäßig vom Backend aktualisiert. Wenn sie schon hier abgelegt werden, dann würde ich sie zumindest in einem Unterobjekt zusammenfassen.
```python
@dataclass
class GeneralVehicleConfig:
    efficiency: float = 90
    use_soc_from_cp: bool = False
    request_interval_charging: int = 5
    request_interval_not_charging: int = 720
    request_only_plugged: bool = False
    manual_soc_state = {
        imported_start: Optional[float] = 0  # don't show in UI
        manual_soc: Optional[int] = None  # don't show in UI
        request_start_soc: bool = True  # don't show in UI
        soc_start: float = 0  # don't show in UI
    }
```
Besser und sauberer wäre es jedoch, die Bewegungsdaten komplett zu trennen.